src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -402,22 +402,37 @@ void SQLiteDBEngine::deleteRowsByStatusField(const nlohmann::json& tableNames)
 }
 
 void SQLiteDBEngine::returnRowsMarkedForDelete(const nlohmann::json& tableNames,
-                                               const DbSync::ResultCallback callback)
+                                               const DbSync::ResultCallback callback,
+                                               const nlohmann::json& options)
 {
+    auto allColumns {false};
+
+    const auto itAllColumns {options.find("all_columns")};
+
+    if (options.end() != itAllColumns)
+    {
+        allColumns = itAllColumns->is_boolean() ? itAllColumns.value().get<bool>() : allColumns;
```

Comment:
Might be easier to read if it looked like this:
```suggestion
        allColumns = itAllColumns->is_boolean() ? itAllColumns.value().get<bool>() : false;
```